packages/suite-desktop-api/src/validation.ts

```diff
@@ -0,0 +1,41 @@
+import { RendererChannels } from './api';
+import { SuiteThemeVariant } from './messages';
+import { UnionToTuple } from './methods';
+
+type Primitive = 'boolean' | 'string' | 'number';
+type OptionalPrimitive = Primitive | [Primitive, boolean];
+
+export const isPrimitive = (type: OptionalPrimitive, value: any) => {
```

Comment:
nit: `isPrimitive` and `isObject` could be moved to `utils` package

Reply:
i dont want to have dependency in this package